src/librustdoc/formats/cache.rs

```diff
@@ -289,15 +291,15 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
                     // A crate has a module at its root, containing all items,
                     // which should not be indexed. The crate-item itself is
                     // inserted later on when serializing the search-index.
-                    if item.def_id.index != CRATE_DEF_INDEX {
+                    if item.def_id.index() != CRATE_DEF_INDEX {
```

Comment:
This will also match the first `DefId::Fake` crated for the crate, since the fake indices start at 0. See my other comment for a suggestion as to how to fix this.